### PR TITLE
IW-1710 publish article renames to article-tags queue

### PR DESCRIPTION
**Pull request by jsutterfield**
Ticket: https://wikia-inc.atlassian.net/browse/IW-1710

Publish article renames to a new article-tags exchange. This exchange will be used by discussions to keep its article tags in-sync with the latest article title.

/cc @Wikia/iwing 

relates to https://github.com/Wikia/pandora/pull/6084

### CODE REVIEW

File: extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+use Wikia\Rabbit\ConnectionBase;
+
+class ArticleTagEventsProducer {
+
+    const RENAMED_ROUTING_KEY= 'article-changed.renamed';
+
+    private static $rabbitPublisher;
+
+    public static function onSpecialMovepageAfterMove( MovePageForm $form, Title $oldTitle, Title $newTitle ) {
```
Comment (mszabo-wikia) [inline extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php]: Let's use `TitleMoveComplete` instead. It will handle API moves as well.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: includes/wikia/VariablesBase.php [hunk 2]
```diff
@@ -8931,3 +8931,13 @@
  * @var bool
  */
 $wgEnableEditDraftSavingExt = false;
+
+/**
+ * ArticleTags RabbitMQ configuration.
+ * @see extensions/wikia/articleTagEvents
+ * @var array $wgArticleTagQueue
+ */
+$wgArticleTagQueue = [
```
Comment (rybmat) [inline includes/wikia/VariablesBase.php]: can we name it `$wgArticleTagExchangeConfig`? it does not contain any info about queue ;)

File: extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        global $wgCityId;
+        $message = [
+            'wikiId' => $wgCityId,
+            'articleId' => $newTitle->getArticleID(),
+            'articleTitle' => $newTitle->getPrefixedText()
```
Comment (rybmat) [inline extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php]: should we also send the relativeUrl here and update it in discussion service?

File: extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php (hunk 1 above)
Comment (jsutterfield) [reply extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php]: Good call! Updated: https://github.com/Wikia/app/pull/16785/commits/c9d3313ca72593ef7cb91950830f56b91ed819f0

**Review by rybmat — CHANGES_REQUESTED** [review]

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php (hunk 3 above)
Comment (jsutterfield) [reply extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php]: Yea, good call 👍

File: includes/wikia/VariablesBase.php (hunk 2 above)
Comment (jsutterfield) [reply includes/wikia/VariablesBase.php]: Sure thing 😛

File: extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php (hunk 3 above)
Comment (jsutterfield) [reply extensions/wikia/ArticleTagEvents/ArticleTagEventsProducer.class.php]: https://github.com/Wikia/app/pull/16785/commits/d075aab2fdef1ea797b1112ebb6c6a211bcb0d09

File: includes/wikia/VariablesBase.php (hunk 2 above)
Comment (jsutterfield) [reply includes/wikia/VariablesBase.php]: https://github.com/Wikia/app/pull/16785/commits/c7cce6935cfa894907cbfbb55c380a2ec0bad7cd

**Review by rybmat — APPROVED** [review]

**Pull request merged**